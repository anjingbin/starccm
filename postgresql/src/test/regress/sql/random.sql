--
-- RANDOM
-- Test the random function
--

-- count the number of tuples originally
SELECT count(*) FROM onek;

-- select roughly 1/10 of the tuples
-- Assume that the "onek" table has 1000 tuples
--  and try to bracket the correct number so we
--  have a regression test which can pass/fail
-- - thomas 1998-08-17
SELECT count(*) AS random INTO RANDOM_TBL
  FROM onek WHERE oidrand(onek.oid, 10);

-- select again, the count should be different
INSERT INTO RANDOM_TBL (random)
  SELECT count(*)
  FROM onek WHERE oidrand(onek.oid, 10);

-- now test the results for randomness in the correct range
SELECT random, count(random) FROM RANDOM_TBL
  GROUP BY random HAVING count(random) > 1;

SELECT random FROM RANDOM_TBL
  WHERE random NOT BETWEEN 80 AND 120;

