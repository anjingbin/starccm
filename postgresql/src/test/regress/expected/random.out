--
-- RANDOM
-- Test the random function
--
-- count the number of tuples originally
SELECT count(*) FROM onek;
 count 
-------
  1000
(1 row)

-- select roughly 1/10 of the tuples
-- Assume that the "onek" table has 1000 tuples
--  and try to bracket the correct number so we
--  have a regression test which can pass/fail
-- - thomas 1998-08-17
SELECT count(*) AS random INTO RANDOM_TBL
  FROM onek WHERE oidrand(onek.oid, 10);
-- select again, the count should be different
INSERT INTO RANDOM_TBL (random)
  SELECT count(*)
  FROM onek WHERE oidrand(onek.oid, 10);
-- now test the results for randomness in the correct range
SELECT random, count(random) FROM RANDOM_TBL
  GROUP BY random HAVING count(random) > 1;
 random | count 
--------+-------
(0 rows)

SELECT random FROM RANDOM_TBL
  WHERE random NOT BETWEEN 80 AND 120;
 random 
--------
(0 rows)

